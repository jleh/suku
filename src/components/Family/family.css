.family {
  margin-top: 2em;
  margin-bottom: 1em;
}

.married {
  font-size: 0.75em;
  padding-top: 0.25em;
}
