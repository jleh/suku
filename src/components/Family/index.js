import React from 'react';
import styled from 'styled-components';
import { Translate } from 'react-localize-redux';

import LinkedPerson from '../LinkedPerson';

const Family = styled.div`
  margin-top: 2em;
  margin-bottom: 1em;
`;

export default ({ families, persons }) => (
  <Family>
    {families.map(family => (
      <div key={family.spouse}>
        <Translate id="family.spouse" />: <LinkedPerson personRef={family.spouse} persons={persons} />
        <ol>
          {family.children.map((child, index) => (
            <li key={index}>
              <LinkedPerson personRef={child} persons={persons} />
            </li>
          ))}
        </ol>
      </div>
    ))}
  </Family>
);
