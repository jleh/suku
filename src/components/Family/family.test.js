import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import Family from '.';

const persons = new Map();

persons.set('H1', {
  handle: 'H1',
  id: '1',
  name: 'Test Person',
  events: {
    birth: '1.1.1900',
    death: '1.1.1950'
  }
});
persons.set('H2', {
  handle: 'H2',
  id: '2',
  name: 'Other Person',
  events: {}
});

const renderComponent = families => renderer
  .create(
    <MemoryRouter>
      <Family families={families} persons={persons} />
    </MemoryRouter>
  )
  .toJSON();

test('Renders family', () => {
  const families = [
    {
      spouse: 'H2',
      children: ['H1']
    }
  ];
  expect(renderComponent(families)).toMatchSnapshot();
});
