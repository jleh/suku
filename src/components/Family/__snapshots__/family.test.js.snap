// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders family 1`] = `
<div
  className="family"
>
  <div>
    Missing translationId: family.spouse for language: \${ languageCode }
    : 
    <div>
      <a
        href="/person/2"
        onClick={[Function]}
      >
        Other Person
      </a>
       
      <span
        className="dates"
      >
         
      </span>
    </div>
    <ol>
      <li>
        <div>
          <a
            href="/person/1"
            onClick={[Function]}
          >
            Test Person
          </a>
           
          <span
            className="dates"
          >
            * 1.1.1900
             
            â€  1.1.1950
          </span>
        </div>
      </li>
    </ol>
  </div>
</div>
`;
